static/scenes/railroad/js/present.js

```diff
@@ -1,28 +1,37 @@
 goog.provide('app.Present');
 
-const material0 = new THREE.MeshToonMaterial( {color: 0xF9D231}); 
+const material0 = new THREE.MeshToonMaterial( {color: 0xF9D231});
+const loader = new THREE.OBJLoader();
+
+let loadedObj;
 
 class Present {
-    constructor(loader, scene, giftWrapMaterial) {
+
+    static async preload() {
+        loadedObj = await loader.loadAsync("models/gift.obj");
+    }
+
+    constructor(scene, giftWrapMaterial) {
+        if (loadedScene == undefined) {
+            throw 'Must call Present.preload() before constructing instance.'
+        }
```

Comment:
Nit: I think a static factory function is a more standard style, at least coming from the old C++ development I did. Would that pattern work here?

Reply:
We could but I think a regular constructor works better here. This check is almost just documentation, as we will always call preload anyway.